generated/php/google-cloud-speech-v1/tests/unit/Speech/V1/SpeechClientTest.php

```diff
@@ -301,4 +304,89 @@ public function longRunningRecognizeExceptionTest()
         $this->assertTrue($grpcStub->isExhausted());
         $this->assertTrue($operationsStub->isExhausted());
     }
+
+    /**
+     * @test
+     */
+    public function streamingRecognizeTest()
+    {
+        $grpcStub = $this->createStub([$this, 'createMockSpeechImpl']);
+        $client = $this->createClient('createSpeechStubFunction', $grpcStub);
+
+        $this->assertTrue($grpcStub->isExhausted());
+
+        // Mock response
+        $expectedResponse = new StreamingRecognizeResponse();
+        $grpcStub->addResponse($expectedResponse);
+        $grpcStub->addResponse($expectedResponse);
+        $grpcStub->addResponse($expectedResponse);
+
+        // Mock request
+        $request = new StreamingRecognizeRequest();
+
+        $bidi = $client->streamingRecognize();
+        $this->assertInstanceOf(BidiStream::class, $bidi);
+
+        $bidi->write($request);
+        $responses = [];
+        $responses[] = $bidi->read();
+
+        $bidi->writeAll([$request, $request]);
+        foreach ($bidi->closeWriteAndReadAll() as $response) {
+            $responses[] = $response;
+        }
+
+        $this->assertEquals([$expectedResponse, $expectedResponse, $expectedResponse], $responses);
+
+        $createStreamRequests = $grpcStub->getReceivedCalls();
+        $this->assertSame(1, count($createStreamRequests));
+        $streamFuncCall = $createStreamRequests[0]->getFuncCall();
+        $streamRequestObject = $createStreamRequests[0]->getRequestObject();
+        $this->assertSame('/google.cloud.speech.v1.Speech/StreamingRecognize', $streamFuncCall);
+        $this->assertNull($streamRequestObject);
+
+        $callObjects = $grpcStub->getCallObjects();
+        $this->assertSame(1, count($callObjects));
+        $bidiCall = $callObjects[0];
+
+        $writeRequests = $bidiCall->getReceivedCalls();
+        $this->assertSame(3, count($writeRequests));
+        foreach ($writeRequests as $writeRequest) {
+        }
```

Comment:
I think we can get rid of this foreach block or is it something going wrong here?